### PR TITLE
Skip dead conditions in eventpipe srcgen

### PR DESCRIPTION
**Pull request by am11**
An example of before/after diff:
```diff
--- artifacts/obj/coreclr/Linux.x64.Debug/vm/eventing/eventpipe/eventpipe/dotnetruntimeprivate.cpp
+++ artifacts/obj/coreclr/Linux.x64.Debug/vm/eventing/eventpipe/eventpipe/dotnetruntimeprivate.cpp

extern "C" ULONG EventPipeWriteEventExecExeEnd(
    LPCGUID ActivityId,
    LPCGUID RelatedActivityId)
{
    if (!EventPipeEventEnabledExecExeEnd())
        return ERROR_SUCCESS;

+     size_t size = 32;
+     BYTE stackBuffer[32];
-    BYTE stackBuffer[32];
    BYTE *buffer = stackBuffer;
    size_t offset = 0;
-    size_t size = 32;
-    bool fixedBuffer = true;
-    bool success = true;


-
-    if (!success)
-    {
-        if (!fixedBuffer)
-            delete[] buffer;
-        return ERROR_WRITE_FAULT;
-    }

    EventPipeAdapter::WriteEvent(EventPipeEventExecExeEnd, (BYTE *)buffer, (unsigned int)offset, ActivityId, RelatedActivityId);
-
-    if (!fixedBuffer)
-        delete[] buffer;

    return ERROR_SUCCESS;
}
```

This is to cleanup the bulk of warnings from gcc 11 by commenting out https://github.com/dotnet/runtime/blob/ef857626c5a0ad29cffb61a2f8db87e083ca2332/eng/native/configurecompiler.cmake#L402-L404 so we can see the real `-Werror=mismatched-new-delete` in coreclr/vm that is causing #56888 when `DOTNET_EnableEventLog` is enabled with debug `corerun`, e.g.

```sh
[ 29%] Building CXX object vm/wks/CMakeFiles/cee_wks_core.dir/__/clsload.cpp.o
/runtime/src/coreclr/vm/ceeload.cpp: In member function 'void Module::InitializeDynamicILCrst()':
/runtime/src/coreclr/vm/ceeload.cpp:1442:16: error: 'void operator delete(void*)' called on pointer returned from a mismatched allocation function [-Werror=mismatched-new-delete]
 1442 |         delete pCrst;
      |                ^~~~~
/runtime/src/coreclr/vm/ceeload.cpp:1438:97: note: returned from 'static void* Crst::operator new(size_t)'
 1438 |     Crst * pCrst = new Crst(CrstDynamicIL, CrstFlags(CRST_UNSAFE_ANYMODE | CRST_DEBUGGER_THREAD));
      |                                                                                                 ^
```

### CODE REVIEW

**Comment by am11** [comment]
cc @josalem, @janvorli

File: src/coreclr/scripts/genEventPipe.py [hunk 1]
```diff
@@ -382,14 +400,22 @@ def generateWriteEventBody(template, providerName, eventName, runtimeFlavor):
...
     footer = ""
-    if runtimeFlavor.coreclr:
-        footer = """
+    if emittedWriteToBuffer:
+        if runtimeFlavor.coreclr:
+            footer = """
     if (!fixedBuffer)
         delete[] buffer;
+
+"""
```
Comment (josalem) [inline src/coreclr/scripts/genEventPipe.py:411]: Couldn't we skip emitting this at all if `fixedBuffer` is already known to be true?

**Review by josalem — APPROVED** [review]

File: src/coreclr/scripts/genEventPipe.py (hunk 1 above)
Comment (am11) [reply src/coreclr/scripts/genEventPipe.py:411]: Yup, and in C (mono) code, we are passing it by pointer so both caller and calllee indicate the linkage (like C#'s ref/out..), but in C++ it's passed by reference so call-sites does not immediately tell what's going to happen with each argument..

ps - I personally prefer using pointers in both C and C++ to keep it explicit both on caller and callee, but I didn't make that change here. 😅

**Comment by josalem** [comment]
Thanks for this PR @am11!

**Pull request merged**